### ISSUE TITLE
[Feature] <title> Simply add file names to side-by-side images

### ISSUE THREAD

**Issue by Digita1Menace** [olive-groves/butterfly_viewer#68]
### Is there an existing request for this feature?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

I use it to drag and drop 2 images to compare them side by side. But sometime they are so similar, I dont know which is which. please ass the filenames somewhere on the screen. thanks.

### Describe the solution you'd like

image names

### Describe alternatives you've considered

xmpview but it's not as sreamlined

### Anything else?

no
![butterfly_viewer_TgxJ922bL3](https://github.com/user-attachments/assets/8b816477-16ee-40a4-a571-578e248b4b4a)

**Comment by larsmaxfield** [comment]
Hi @Digita1Menace, the filename(s) of the active image window are shown when the interface is toggled on ("showing") like this:

![image](https://github.com/user-attachments/assets/a6e49667-c460-4050-8966-19eed9067b5a)

But I admit that's not ideal; I've been meaning to improve how filenames are shown.

I've also thought of letting users selectively toggle the visibility of each interface element — filenames, the sliding overlay creator panel, the bottom-right toolbar.

Do you have any ideas on that?

**Comment by Digita1Menace** [comment]
I'm not a programmer. The interface is in the way when comparing two screenshots. I encode movies and also upgrade movies. I use BV for this only. maybe can you just make slimmed down version with no interface besides what's in the bottom-right? plus have the filenames? I did discover though that BV will load images in alphabetical order though, but this forces me to rename one of the files before loading both into it.

**Comment by Digita1Menace** [comment]
I just want a side by side compare with the file name above, and slightly bigger font on the file names. that would be ideal
![Uploading Viewer screenshot 2024-08-16 094712.png…]()

**Comment by Digita1Menace** [comment]
![Viewer screenshot 2024-08-16 094712](https://github.com/user-attachments/assets/881868ff-de20-4389-a8dc-0266811d3bd2)

**Comment by Digita1Menace** [comment]
Check out this compare tool. it is for the browser. A standalone of this is perfect. drag two images into it.
https://tshino.github.io/compare/compare.html

**Comment by larsmaxfield** [comment]
> maybe can you just make slimmed down version with no interface besides what's in the bottom-right [...] plus have the filenames?

Good point. Maybe have two toggles — show/hide the tools and show/hide the file metadata, like how Adobe Lightroom does it.

> BV will load images in alphabetical order [...] this forces me to rename one of the files before loading both into it.

If you want a set of images in a particular order when arranged left-to-right or top-to-bottom, you can individually drag them into the Viewer and they'll be in that order.

> slightly bigger font on the file names. that would be ideal

Noted!

**Comment by larsmaxfield** [comment]
That compare tool is really neat.

I was able to get it working locally as a standalone app. Did you try unzipping the contents and then double-clicking the `compare.html`? That's how I get it working on my computer:

![image](https://github.com/user-attachments/assets/aa770e0d-00ae-4397-a429-66a5ff5c492f)

**Comment by Digita1Menace** [comment]
Settings would be nice too. like remember background color, remember display size of images, ect. as of now, every time I do a compare, I have to all these steps

https://github.com/user-attachments/assets/6fc1aa80-5bed-4fbd-b711-08e93e9f42cd

**Comment by Digita1Menace** [comment]
Yes, I tried that, on both my hardened fiefox and ungoogled chromium, I get this: may it would work with a normal version of either of those, but I'm not switching just for that tool, lol
![chrome_ox9bjkqXlX](https://github.com/user-attachments/assets/0c9b57fe-a04c-4587-ba3b-4361d43c92dd)

**Comment by Digita1Menace** [comment]
and I know windows 11 does this smoothly, but I dont use that crap. Wind 10 LTSC so I cant do quick compares like Win 11 can

**Comment by Digita1Menace** [comment]
how about just 1 extra button that does all the steps in my video, plus the files overlayed in the top left of the image, slightly transparent? thats ideal. thanks.

**Comment by larsmaxfield** [comment]
Good point on the settings. I made a separate feature request for that just now: https://github.com/olive-groves/butterfly_viewer/issues/69

**Comment by Digita1Menace** [comment]
You made a feature request on your own program? lol, IDK I guess I don't how GitHub works too good.

**Comment by larsmaxfield** [comment]
Yes, it keeps track of my own to-do's and lets people know what I've got in mind — I'm also myself a user!

**Comment by larsmaxfield** [comment]
Hmm, either a button or even a shortcut key for those steps.

To make sure I understand the steps, do you mean that (upon dragging in two images) the button/shortcut should trigger this?
- Turn off interface (but show filenames)
- Fullscreen
- Zoom to fit

**Comment by Digita1Menace** [comment]
Got it. oh yea, also remember to switch all sampling off. I know it remembers sampling on or off for individual images, but I am not sure if it remember ALL OFF. does it?

**Comment by Digita1Menace** [comment]
> Hmm, either a button or even a shortcut key for those steps would be possible.
> 
> To make sure I understand the steps, do you mean that (upon dragging in two images) the button/shortcut should trigger this?
> 
>     * Turn off interface (but show filenames)
> 
>     * Fullscreen
> 
>     * Zoom to fit

Yea, so after I drag two images, I click a button and it does all that.

**Comment by larsmaxfield** [comment]
On startup, upsampling is always ALL OFF — it indeed doesn't remember if ALL OFF. I'll add that to settings wishlist.

**Comment by Digita1Menace** [comment]
Your right, it does remember. one last thing, after I close fullscreen when done comparing, I still have another window open I have to also close
![butterfly_viewer_zqmXU1Vv7k](https://github.com/user-attachments/assets/1ab79bec-4075-4dab-969f-be790ca86483)

**Comment by larsmaxfield** [comment]
Ah yes, that's a relic of solving problems I experienced when implementing fullscreen mode. Something with multiple monitors, or (dis)connecting monitors, or moving/closing windows, or a combination of those...

To avoid that you could first exit fullscreen with the `Esc` key, and then close the Viewer.

**Comment by Digita1Menace** [comment]
got it. thanks!

**Comment by larsmaxfield** [comment]
What if, instead of a new button or shortcut, the Viewer simply does all those steps when you click fullscreen after having dragged in the images. It would check if you had panned or zoomed — if not, then it would assume those steps.

Fullscreen by default could also trigger hiding the interface, as that's what YouTube and other apps assume when you go fullscreen.

**Comment by Digita1Menace** [comment]
And not entirely necessary, but like I mentioned, a semi translucent file title OVERLAYED in the top-left of the image, not above it, that way the image are in complete full screen, not slightly smaller to accommodate the title.s space.
just like that web app does
![firefox_UOpxrAudph](https://github.com/user-attachments/assets/085c9a0d-3cfc-4139-993b-106fdb6bb255)

**Comment by larsmaxfield** [comment]
Makes sense. Thanks for the feedback and tips! I'll update here when there's progress.
